typescript/token/test/HypNativeScaled.t.sol

```diff
@@ -0,0 +1,148 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity >=0.8.0;
+
+import "forge-std/Test.sol";
+
+import {HypNativeScaled} from "../contracts/extensions/HypNativeScaled.sol";
+import {HypERC20} from "../contracts/HypERC20.sol";
+import {TypeCasts} from "@hyperlane-xyz/core/contracts/libs/TypeCasts.sol";
+import {MockHyperlaneEnvironment} from "@hyperlane-xyz/core/contracts/mock/MockHyperlaneEnvironment.sol";
+
+contract HypNativeScaledTest is Test {
+    uint32 nativeDomain = 1;
+    uint32 synthDomain = 2;
+
+    uint8 decimals = 9;
+    uint256 scale = 10**9;
+    uint256 synthSupply = 123456789;
+
+    event Donation(address indexed sender, uint256 amount);
+    event SentTransferRemote(
+        uint32 indexed destination,
+        bytes32 indexed recipient,
+        uint256 amount
+    );
+    event ReceivedTransferRemote(
+        uint32 indexed origin,
+        bytes32 indexed recipient,
+        uint256 amount
+    );
+
+    HypNativeScaled native;
+    HypERC20 synth;
+
+    MockHyperlaneEnvironment environment;
+
+    function setUp() public {
+        environment = new MockHyperlaneEnvironment(synthDomain, nativeDomain);
+
+        synth = new HypERC20(decimals);
+        synth.initialize(
+            address(environment.mailboxes(synthDomain)),
+            address(environment.igps(synthDomain)),
+            synthSupply,
+            "Zebec BSC Token",
+            "ZBC"
+        );
+
+        native = new HypNativeScaled(scale);
```

Comment:
This is assuming 18 decimals for the token, and i would write the test cases to be `amount`/user-focused